x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -322,17 +320,30 @@ else if (exp instanceof GroupingFunction) {
                             // date histogram
                             if (isDateBased(h.dataType())) {
                                 Object value = h.interval().value();
-                                // interval of exactly 1 year
-                                if (value instanceof IntervalYearMonth
-                                        && ((IntervalYearMonth) value).interval().equals(Period.ofYears(1))) {
-                                    String calendarInterval = Year.YEAR_INTERVAL;
+                                // interval of exactly 1 year or 1 month
+                                if (value instanceof IntervalYearMonth &&
+                                        (((IntervalYearMonth) value).interval().equals(Period.ofYears(1)) 
+                                        || ((IntervalYearMonth) value).interval().equals(Period.ofMonths(1)))) {
+                                    Period yearMonth = ((IntervalYearMonth) value).interval();
+                                    
+                                    String calendarInterval = null;
+                                    if (yearMonth.equals(Period.ofYears(1))) {
+                                        calendarInterval = Histogram.YEAR_INTERVAL;
+                                    } else if (yearMonth.equals(Period.ofMonths(1))) {
+                                        calendarInterval = Histogram.MONTH_INTERVAL;
+                                    }
 
-                                    // When the histogram is `INTERVAL '1' YEAR`, the interval used in the ES date_histogram will be
-                                    // a calendar_interval with value "1y". All other intervals will be fixed_intervals expressed in ms.
-                                    if (field instanceof FieldAttribute) {
-                                        key = new GroupByDateHistogram(aggId, QueryTranslator.nameOf(field), calendarInterval, h.zoneId());
-                                    } else if (field instanceof Function) {
-                                        key = new GroupByDateHistogram(aggId, ((Function) field).asScript(), calendarInterval, h.zoneId());
+                                    // When the histogram is `INTERVAL '1' YEAR` or `INTERVAL '1' MONTH`, the interval used in 
+                                    // the ES date_histogram will be a calendar_interval with value "1y" or "1M" respectively.
+                                    // All other intervals will be fixed_intervals expressed in ms.
+                                    if (calendarInterval != null) {
```

Comment:
Here too, the null check is superfluous as the whole block is guarded by: 
```
&& (((IntervalYearMonth) value).interval().equals(Period.ofYears(1)) || ((IntervalYearMonth) value).interval().equals(Period.ofMonths(1)))
```

Reply:
Yep, true. It's a leftover from a previous idea.